### PR TITLE
feat: Upgradeability and access control

### PR DESCRIPTION
**Pull request by ermvrs**
- [ ] issue #https://github.com/keep-starknet-strange/blobstream-starknet/issues/52
- [x] follows contribution [guide](../CONTRIBUTING.md)
- [x] code change includes tests
- [ ] breaking change

Added Openzeppelins ownable and upgradeability components. Upgradeability was added as a new interface to increase reusability. Tests implemented

### CODE REVIEW

File: src/lib.cairo [hunk 1]
```diff
@@ -65,14 +65,32 @@ trait IDAOracle<TContractState> {
     );
 }
 
+#[starknet::interface]
```
[the remark this answers is not in the document]
Comment (ermvrs) [reply src/lib.cairo:68]: Both IDAO and IUpgradeable moved

File: src/tests/test_blobstream.cairo [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+    IUpgradeableDispatcherTrait
 };
 use core::bytes_31::one_shift_left_bytes_u128;
-use snforge_std::{declare, ContractClassTrait};
-use starknet::EthAddress;
+use snforge_std::{declare, ContractClassTrait, start_prank, stop_prank, CheatTarget};
 use starknet::secp256_trait::Signature;
+use starknet::{EthAddress, ContractAddress, ClassHash};
+
+fn OWNER() -> ContractAddress {
```
[the remark this answers is not in the document]
Comment (ermvrs) [reply src/tests/test_blobstream.cairo:12]: done

**Review by drspacemn — APPROVED** [review]
beautiful thank you!

**Comment by drspacemn** [comment]
@all-contribs add @ermvrs for code

**Comment by drspacemn** [comment]
@all-contributors add @ermvrs for code

**Comment by drspacemn** [comment]
https://github.com/all-contributors add @ermvrs for code

**Comment by drspacemn** [comment]
@all-contributors add @ermvrs for code

**Pull request merged**